### PR TITLE
Runtime reports correct atom type

### PR DESCRIPTION
**Pull request by 4e6**
### Pull Request Description

Fix the Atom type reported by the runtime. I.e. the type of `Some 42` expression is now `Maybe`, and not `Some`.

### Checklist

Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the
      [Scala](https://github.com/enso-org/enso/blob/develop/docs/style-guide/scala.md),
      [Java](https://github.com/enso-org/enso/blob/develop/docs/style-guide/java.md),
      and
      [Rust](https://github.com/enso-org/enso/blob/develop/docs/style-guide/rust.md)
      style guides.
- All code has been tested:
  - [x] Unit tests have been written where possible.
  - [ ] ~If GUI codebase was changed: Enso GUI was tested when built using BOTH
        `./run ide build` and `./run ide watch`.~

### CODE REVIEW

File: engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/AtomConstructor.java [hunk 1]
```diff
@@ -32,7 +32,7 @@
 public final class AtomConstructor implements TruffleObject {
 
   private final String name;
-  private @CompilerDirectives.CompilationFinal ModuleScope definitionScope;
+  private final @CompilerDirectives.CompilationFinal ModuleScope definitionScope;
```
Comment (hubertp) [inline engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/AtomConstructor.java]: If you make it final then
```suggestion
  private final ModuleScope definitionScope;
```

**Review by hubertp — COMMENTED** [review]
nit

File: engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/AtomConstructor.java (hunk 1 above)
Comment (JaroslavTulach) [reply engine/runtime/src/main/java/org/enso/interpreter/runtime/callable/atom/AtomConstructor.java]: Yes, `final` supersedes `@CompilationFinal`.

**Review by JaroslavTulach — APPROVED** [review]

File: test/Tests/src/Semantic/Meta_Location_Spec.enso [hunk 2]
```diff
@@ -21,7 +21,7 @@ spec = Test.group "Meta-Value Inspection" <|
         x = 42
         y = My_Type.Value 1 2 3
         Meta.get_qualified_type_name x . should_equal "Standard.Base.Data.Numbers.Integer"
-        Meta.get_qualified_type_name y . should_equal "enso_dev.Tests.Semantic.Meta_Location_Spec.My_Type.Value"
+        Meta.get_qualified_type_name y . should_equal "enso_dev.Tests.Semantic.Meta_Location_Spec.My_Type"
```
Comment (4e6) [inline test/Tests/src/Semantic/Meta_Location_Spec.enso:24]: @jdunkerley now `Meta.get_qualified_type_name` also reports the name of a base type, not constructor

**Review by hubertp — APPROVED** [review]

**Review by jdunkerley — APPROVED** [review]

**Pull request merged**